Add quality of budget institution narrative & plots using PEFA data

weilu opened the pull request:
Depends on https://github.com/dime-worldbank/mega-indicators/pull/11

bhupatiraju commented on components/pefa.py:
```diff
@@ -0,0 +1,268 @@
+    (1, "D"), (1.5, "D+"),
+])
+
+PILLAR_MAPPING = OrderedDict([
```
In the narrative it reads better to have something like "the strongest pillar of Bangladesh’s budget institutions is "Policy-based budgeting" rather than "the strongest pillar of Bangladesh’s budget institutions is "4. Policy-based budgeting".

So it might make sense to remove the leading numbers in the name of the PEFA categories, unless the audience would know and find it helpful to know which numbered category it is.

weilu replied:
I'm on the fence about this one as the leading number makes it easy to identify the corresponding pillar in the heatmap quickly, but I removed them because you are right, it reads better this way.

bhupatiraju commented on components/pefa.py:
```diff
@@ -0,0 +1,268 @@
+            'xanchor': 'center',
+        },
+        plot_bgcolor="white",
+        legend=dict(orientation="h", yanchor="bottom", y=1.03),
```
x=0.5 would centre the legend under the legend title? It would also be good to have a little space between the legend title and the legend bar

weilu replied:
I assume you are talking about the heatmap legend, not the overall scatter plot? I've moved the heatmap title to the side to be consistent with the scatter plot and make the top less crowded. Let me know what you think.

bhupatiraju reviewed (approved):
Looks good!

Couple of points:
- The assessment and liability cell in the heatmap for years before 2016 has hover data but is not being colored. For example Albania has a value 3.1 for this but is not being show the appropriate color. Not sure if this is an issue at my end but this issue persists for other countries as well.
- For Uruguay where we have no data we should probably not display the empty plots. Just the text saying that we don't have data should suffice. 
- [Minor issue] when there is only one data point from PEFA the x-ticks should probably say on the year and not something like 2016.2, 2016.4 etc

weilu commented:
> * The assessment and liability cell in the heatmap for years before 2016 has hover data but is not being colored. For example Albania has a value 3.1 for this but is not being show the appropriate color. Not sure if this is an issue at my end but this issue persists for other countries as well.

Good catch! There was actually a mismatch between grades and scores. I reversed the index for one but left the other. I fixed by sorting both and specifically handle missing values. Before NaN score was mapped to "A" (oops), now it says "N/A" for both score and grade, when missing.

> * For Uruguay where we have no data we should probably not display the empty plots. Just the text saying that we don't have data should suffice.

Done

> * [Minor issue] when there is only one data point from PEFA the x-ticks should probably say on the year and not something like 2016.2, 2016.4 etc

Fixed
